modin/numpy/arr.py

```diff
@@ -0,0 +1,333 @@
+            self._query_compiler = qc
+            self._ndim = 1
+        else:
+            expected_kwargs = {
+                "dtype": None,
+                "copy": True,
+                "order": "K",
+                "subok": False,
+                "ndmin": 0,
+                "like": None,
+            }
+            rcvd_kwargs = {
+                "dtype": dtype,
+                "copy": copy,
+                "order": order,
+                "subok": subok,
+                "ndmin": ndmin,
+                "like": like,
+            }
+            for key, value in rcvd_kwargs.copy().items():
+                if value == expected_kwargs[key]:
+                    rcvd_kwargs.pop(key)
+            arr = numpy.array(object, **rcvd_kwargs)
+            self._ndim = len(arr.shape)
+            if self._ndim > 2:
+                ErrorMessage.not_implemented(
+                    "NumPy arrays with dimensions higher than 2 are not yet supported."
+                )
+            import modin.pandas as pd
+
+            self._query_compiler = pd.DataFrame(arr)._query_compiler
+
+    def _absolute(
+        self,
+        out=None,
+        where=True,
+        casting="same_kind",
+        order="K",
+        dtype=None,
+        subok=True,
+    ):
+        result = self._query_compiler.abs()
+        return array(_query_compiler=result, _ndim=self._ndim)
+
+    __abs__ = _absolute
+
+    def _binary_op(self, other):
+        broadcast = self._ndim != other._ndim
+        if broadcast:
+            # In this case, we have a 1D object doing a binary op with a 2D object
+            caller = self if self._ndim == 2 else other
+            callee = other if self._ndim == 2 else self
+            return (caller, callee, caller._ndim, {"broadcast": broadcast, "axis": 1})
+        else:
+            if self.shape != other.shape:
+                # In this case, we either have two mismatched objects trying to do an operation
+                # or a nested 1D object that must be broadcasted trying to do an operation.
+                matched_dimension = None
+                if self.shape[0] == other.shape[0]:
+                    matched_dimension = 0
+                elif self.shape[1] == other.shape[1]:
+                    matched_dimension = 1
+                if matched_dimension is not None:
+                    if (
+                        self.shape[matched_dimension ^ 1] == 1
+                        or other.shape[matched_dimension ^ 1] == 1
+                    ):
+                        # caller = self if other.shape[matched_dimension ^ 1] == 1 else other
+                        # callee = other if other.shape[matched_dimension ^ 1] == 1 else self
+                        return (self, other, self._ndim, {"broadcast": True, "axis": 1})
+                else:
+                    raise ValueError(
+                        f"operands could not be broadcast together with shapes {self.shape} {other.shape}"
+                    )
```

Comment:
```suggestion
                if self.shape[0] == other.shape[0]:
                    matched_dimension = 0
                elif self.shape[1] == other.shape[1]:
                    matched_dimension = 1
                else:
                    raise ValueError(
                        f"operands could not be broadcast together with shapes {self.shape} {other.shape}"
                    )

                if (
                    self.shape[matched_dimension ^ 1] == 1
                    or other.shape[matched_dimension ^ 1] == 1
                ):
                    # caller = self if other.shape[matched_dimension ^ 1] == 1 else other
                    # callee = other if other.shape[matched_dimension ^ 1] == 1 else self
                    return (self, other, self._ndim, {"broadcast": True, "axis": 1})
```

Reply:
Not sure if we wanted to leave this case for later, but this code doesn't work when one of the dimensions is 1, e.g. `np.array([[1] * 5] * 4) * np.array([2])` (a 5x4 array multiplied by a 1x1 array). Per numpy docs:
> When operating on two arrays, NumPy compares their shapes element-wise. It starts with the trailing (i.e. rightmost) dimension and works its way left. Two dimensions are compatible when
> 1. they are equal, or
> 2. one of them is 1.